src/rule.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::Arc;
+
+use derive_builder::Builder;
+use tree_sitter::{Node, Query};
+
+use crate::context::{Context, QueryMatchContext};
+
+#[derive(Builder)]
+#[builder(setter(into))]
+pub struct Rule {
+    pub name: String,
+    #[builder(setter(custom))]
+    pub create: Arc<dyn Fn(&Context) -> Vec<RuleListener>>,
```

Comment:
Ah, not sure I realized you could directly contain a `dyn` trait in an `Arc`, but that makes sense, since it's essentially a fancy `Box`?

Reply:
Yup and `Box<dyn Trait>`/`Rc<dyn Trait>`/`Arc<dyn Trait>` are all "special" as far as being able to be the "`self` receiver type", I take advantage of that in a subsequent PR